raiden/transfer/architecture.py

```diff
@@ -272,6 +295,66 @@ def dispatch(self, state_change: StateChange) -> List[Event]:
 
         return events
 
+    def detect_balance_proof_change(self, old_state, current_state):
+        """Compare two states for any changed received balance_proofs"""
+        if old_state == current_state:
+            return
+        # Assuming, there is no transition None->new balance proof:
+        if not hasattr(old_state, 'identifiers_to_paymentnetworks'):
```

Comment:
Checking if the attribute exists is not sufficient, you should check if it's not `None`, `getattr(.., .., None)` is probably best here.